lib/rubocop/cop/rspec/http_status.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'symbol_to_status_code'
+
+module RuboCop
+  module Cop
+    module RSpec
+      # This cop enforces consistent use of `symbol` or `numeric`
+      # value to describe http status.
+      #
+      # @example `EnforcedStyle: symbol` (default)
+      #   # bad
+      #   it { is_expected.to have_http_status 200 }
+      #   it { is_expected.to have_http_status 404 }
+      #
+      #   # good
+      #   it { is_expected.to have_http_status :ok }
+      #   it { is_expected.to have_http_status :not_found }
+      #
+      #
+      # @example `EnforcedStyle: numeric`
+      #   # bad
+      #   it { is_expected.to have_http_status :ok }
+      #   it { is_expected.to have_http_status :not_found }
+      #
+      #   # good
+      #   it { is_expected.to have_http_status 200 }
+      #   it { is_expected.to have_http_status 404 }
+      #
+      class HttpStatus < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<prefer>s` over `%<current>s` '\
+              'to describe http status code.'.freeze
+
+        def_node_matcher :http_status, <<-PATTERN
+          (send _ :have_http_status ${int sym})
+        PATTERN
+
+        def on_send(node)
+          http_status(node) do |value_type|
+            return if style == :numeric && value_type.int_type?
+            return if style == :symbol && value_type.sym_type?
+
+            add_offense(node)
+          end
+        end
+
+        def message(node)
+          if node.each_descendant.find(&:sym_type?)
+            format(MSG, prefer: 'numeric', current: 'symbol')
+          else
+            format(MSG, prefer: 'symbol', current: 'numeric')
+          end
+        end
+
+        def autocorrect(node)
+          replacement = new_value(node.children[2])
+          return if replacement.nil?
+
+          lambda do |corrector|
+            replacement = "#{node.children[1]} #{replacement}"
+            corrector.replace(node.loc.expression, replacement)
```

Comment:
if you use node.loc.selector, I guess you don't need to build the replacement with the node.children[1], but just use the new_value